libs/base/Data/Vect.idr

```diff
@@ -87,6 +87,14 @@ take : (n  : Nat)
 take 0 xs = Nil
 take (S k) (x :: xs) = x :: take k xs
 
+namespace Stream
+  ||| Take precisely n elements from the stream.
+  ||| @ n how many elements to take
+  ||| @ xs the stream
+  public export
+  take : (n : Nat) -> (xs : Stream a) -> Vect n a
+  take n xs = fromList $ take n xs
```

Comment:
i'm not clear on the implications of using `fromList` instead of copy-pasting from the `List` implementation. Anyone know?

Reply:
In scheme `fromList` should essentially be the identity function given that both
`List` and `Vect` will be represented using scheme lists. I don't know whether either
the backend of Chez detects that and optimises it away though.

Reply:
`fromList` is defined later so I can't use it here. Since it's not clear which is better, I just copied the implementation from the `List` version